iOS_SDK/OneSignalSDK/Source/OSMessagingController.m

```diff
@@ -328,14 +481,19 @@ - (void)messageViewDidSelectAction:(OSInAppMessage *)message withAction:(OSInApp
     if (self.actionClickBlock)
         self.actionClickBlock(action);
   
+    let clickId = action.clickId;
+    BOOL clickAvailableByRedisplay = [message.displayStats isRedisplayEnabled] && [message isClickAvailable:clickId];
+    
     // Make sure no click tracking is performed for IAM previews
     // If the IAM clickId exists within the cached clickedClickIds return early so the click is not tracked
+    // unless that click is from an IAM with redisplay
     // Handles body, button, or image clicks
-    if (message.isPreview || [self.clickedClickIds containsObject:action.clickId])
+    if (message.isPreview || (!clickAvailableByRedisplay && [self.clickedClickIds containsObject:clickId]))
         return;
```

Comment:
Don't you clear the `clickedClickIds` when reseting an IAM for redisplay? Why would you need the `clickAvailableByRedisplay` variable?

Reply:
clickedClickIds are from all the IAMs, i needed a way to track the clickIds per IAM in order to clear only the Ids for that IAM. thats why i have this check

Reply:
Added new comments for this